snuba/utils/streams/backends/kafka.py

```diff
@@ -615,26 +619,114 @@ def closed(self) -> bool:
 
 DEFAULT_QUEUED_MAX_MESSAGE_KBYTES = 50000
 DEFAULT_QUEUED_MIN_MESSAGES = 10000
+DEFAULT_PARTITIONER = "consistent"
+DEFAULT_MAX_MESSAGE_BYTES = 50000000  # 50MB, default is 1MB
+SUPPORTED_KAFKA_CONFIGURATION = (
+    # Check https://github.com/edenhill/librdkafka/blob/master/CONFIGURATION.md
+    # for the full list of available options
+    "bootstrap.servers",
+    "sasl.mechanism",
+    "sasl.username",
+    "sasl.password",
+    "security.protocol",
+)
+
+
+def get_default_kafka_configuration(
+    storage_key: Optional[StorageKey] = None,
+    bootstrap_servers: Optional[Sequence[str]] = None,
+    override_params: Optional[Mapping[str, Any]] = None,
+) -> KafkaBrokerConfig:
+    default_bootstrap_servers = None
+    if storage_key is not None:
+        storage_name = storage_key.value
+        if storage_name in settings.DEFAULT_STORAGE_BROKERS:
+            # this is now deprecated
+            logger.warning(
+                "DEPRECATED: DEFAULT_STORAGE_BROKERS is defined. Please use STORAGE_BROKER_CONFIG instead"
+            )
+            default_config = {}
+            default_bootstrap_servers = ",".join(
+                settings.DEFAULT_STORAGE_BROKERS[storage_name]
+            )
+        elif settings.DEFAULT_BROKERS:
+            default_config = {}
+            default_bootstrap_servers = ",".join(settings.DEFAULT_BROKERS)
+        else:
+            default_config = settings.STORAGE_BROKER_CONFIG.get(
+                storage_name, settings.BROKER_CONFIG
+            )
+    else:
+        if settings.DEFAULT_BROKERS:
+            default_config = {}
+            default_bootstrap_servers = ",".join(settings.DEFAULT_BROKERS)
+        else:
+            default_config = settings.BROKER_CONFIG
+    broker_config = copy.deepcopy(default_config)
+    bootstrap_servers = bootstrap_servers or default_bootstrap_servers
+    if bootstrap_servers:
+        broker_config["bootstrap.servers"] = bootstrap_servers
```

Comment:
There is a bug here.
`default_bootstrap_servers` is a string here so everything is fine, though if you pass bootstrap-server to the command line. Now the line that joins the sequence into a string does not exist anymore.
It should be something like:
```
bootrstap_servers = ",".join(bootstrap_servers) if bootstrap_servers else default_bootstrap_servers. 
```

You can test this by removing the config from the settings and running a consumer this way:
```
snuba consumer --storage=events --bootstrap-server=localhost:9092
```

Now the confluent Kafka API receives a sequence instead of a string.

Reply:
Good catch. Let me add a test case and fix this!

Reply:
updated